hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/node/DatanodeAdminMonitorImpl.java

```diff
@@ -87,6 +87,44 @@ public class DatanodeAdminMonitorImpl implements DatanodeAdminMonitor {
   private long unhealthyContainers = 0;
   private long underReplicatedContainers = 0;
 
+  /**
+   * Inner class for snapshot of Datanode ContainerState in
+   * Decommissioning and Maintenance mode workflow.
+   */
+  public static final class TrackedNode {
+
+    private DatanodeDetails datanodeDetails;
+
+    private long startTime = 0L;
+
+    public TrackedNode(DatanodeDetails datanodeDetails) {
```

Comment:
I might be a small improvement to make the constructor take both datanodeDetails and StartTime - then the object can be immutable and we can get rid of the `setStartTime()` method.